fedimint-server/src/config/api.rs

```diff
@@ -755,31 +811,44 @@ mod tests {
                 p2p_url,
                 api_url: api_url.clone(),
             };
-            let server = run_server(
-                data_dir.clone(),
-                our_connections.clone(),
-                db,
-                Default::default(),
-                Default::default(),
-                Default::default(),
-            )
-            .await;
+            let dir = data_dir.join(name_suffix.to_string());
+            fs::create_dir_all(dir.clone()).expect("Unable to create test dir");
+
+            let connections = our_connections.clone();
+            let mut task_group = TaskGroup::new();
+            let subgroup = task_group.make_subgroup().await;
+            task_group
+                .spawn("run-server", move |_| async move {
+                    run_server(
+                        dir,
+                        connections,
+                        db,
+                        Default::default(),
+                        Default::default(),
+                        Default::default(),
+                        subgroup,
+                    )
+                    .await
+                })
+                .await;
             // our id doesn't really exist at this point
             let auth = ApiAuth(format!("password-{port}"));
-            let client = WsAdminClient::new(api_url.clone(), PeerId::from(0), auth.clone());
+            let client = WsAdminClient::new(api_url.clone(), PeerId::from(0), auth);
 
             TestConfigApi {
                 client,
-                server,
-                auth,
+                task_group,
                 name,
                 our_connections,
             }
         }
 
-        /// Helper function using the auth we generated
-        async fn set_password(&self) -> FederationResult<()> {
-            self.client.set_password(self.auth.clone()).await
+        /// Helper function using the auth we generated, retries until the API
```

Comment:
:sus: . Seems like some another `oneshot::channel` called `api_read_receiver` could be better. Especially that we can't tell if something else is wrong, or it is just not ready or what exactly.

Reply:
I didn't want to pollute the production code with code that would only be used for this one test.  In practice, clients will have to retry like this once the API server starts.